From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florida <florida@example.com>
Date: Mon, 01 Jul 2026 00:00:00 +0000
Subject: [PATCH 4/8] Florida: process-thread-name

Randomize process name and main loop thread name to avoid /proc detection.
---
 src/frida-glue.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/frida-glue.c b/src/frida-glue.c
index 6154034..798cf3a 100644
--- a/src/frida-glue.c
+++ b/src/frida-glue.c
@@ -60,7 +60,9 @@ frida_init_with_runtime (FridaRuntime rt)
     {
       main_context = g_main_context_ref (g_main_context_default ());
       main_loop = g_main_loop_new (main_context, FALSE);
-      main_thread = g_thread_new ("frida-main-loop", run_main_loop, NULL);
+      // Florida: randomize main loop thread name
+      g_set_prgname ("ggbond");
+      main_thread = g_thread_new ("ggbond-main", run_main_loop, NULL);
     }

     g_once_init_leave (&frida_initialized, TRUE);
--
2.34.1
